langlib/langlib-test/src/test/resources/test-src/regexp_test.bal

```diff
@@ -1311,6 +1311,17 @@ function testTranslatingDiffNodesInCharClass() {
     assertEquality("A", (<regexp:Span>res8).substring());
 }
 
+function testRegexpWithUnicodeChars() {
+    regexp:Span? resA1 = regexp:find(re = re `\u{1F30D}`, str = "😀 Hello, 🌍! ❤️");
+    assertTrue(resA1 is regexp:Span);
+    regexp:Span? resA2 = regexp:find(re = re `\u{1F31F}`, str = "😀 Hello, 🌍! ❤️");
+    assertTrue(resA2 is ());
+    regexp:Span? resA3 = regexp:find(re = re `\u{0d9c}`, str = "පරිගණක 10");
+    assertTrue(resA3 is regexp:Span);
+    regexp:Span[] resA4 = regexp:findAll(re = re `\u{1F30D}`, str = "😀 Hello, 🌍🌍! ❤️");
+    assertTrue(resA4 is regexp:Span[]);
+}
```

Comment:
We need more tests with assertions of start index and end indexes

Reply:
Also add tests with `CharSetRanges`. And lets have at least two tests pre langlib function.

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/40368/commits/e26daea0c1d5b4a0c11d068f104eda74dd5ce5c5.